src/main/haskell/language-kore/src/Data/Kore/ASTUtils/SmartConstructors.hs

```diff
@@ -0,0 +1,328 @@
+Stability   : experimental
+Portability : portable
+-}
+
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE Rank2Types             #-}
+{-# LANGUAGE AllowAmbiguousTypes    #-}
+{-# LANGUAGE FlexibleContexts       #-}
+{-# LANGUAGE BangPatterns           #-}
+{-# LANGUAGE LambdaCase             #-}
+{-# LANGUAGE DeriveFunctor          #-}
+{-# LANGUAGE DeriveFoldable         #-}
+{-# LANGUAGE DeriveTraversable      #-}
+{-# LANGUAGE TypeFamilies           #-}
+{-# LANGUAGE PatternSynonyms        #-}
+{-# LANGUAGE ScopedTypeVariables    #-}
+{-# LANGUAGE TypeApplications       #-}
+{-# LANGUAGE StandaloneDeriving     #-}
+{-# LANGUAGE TypeSynonymInstances   #-}
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-missing-signatures #-}
+{-# OPTIONS_GHC -Wno-missing-pattern-synonym-signatures #-}
+module Data.Kore.ASTUtils.SmartConstructors where
+
+
+import           Control.Lens
+import           Control.Monad.State
+
+import           Data.Fix
+import           Data.Reflection
+import           Data.Foldable
+
+
+import           Data.Kore.IndexedModule.MetadataTools
+import           Data.Kore.AST.Common
+import           Data.Kore.AST.MLPatterns
+import           Data.Kore.AST.MetaOrObject
+import           Data.Kore.AST.PureML 
+
+getSort 
+  :: (MetaOrObject level, Given (MetadataTools level))
+  => CommonPurePattern level
+  -> Sort level
+getSort x = (getPatternResultSort (getResultSort given) $ unFix x)
+
+flexibleSort 
+  :: MetaOrObject level 
+  => Sort level
+flexibleSort =
+  SortVariableSort $ SortVariable 
+    { getSortVariable = noLocationId "__FlexibleSort__" } --FIXME
+
+pattern And_          s2   a b = Fix (AndPattern (And s2 a b))
+pattern App_ h c               = Fix (ApplicationPattern (Application h c))
+pattern Bottom_       s2       = Fix (BottomPattern (Bottom s2))
+pattern Ceil_      s1 s2   a   = Fix (CeilPattern (Ceil s1 s2 a))
+pattern DV_           s2   a   = Fix (DomainValuePattern (DomainValue s2 a))
+pattern Equals_    s1 s2   a b = Fix (EqualsPattern (Equals s1 s2 a b))
+pattern Exists_       s2 v a   = Fix (ExistsPattern (Exists s2 v a))
+pattern Floor_     s1 s2   a   = Fix (FloorPattern (Floor s1 s2 a))
+pattern Forall_       s2 v a   = Fix (ForallPattern (Forall s2 v a))
+pattern Iff_          s2   a b = Fix (IffPattern (Iff s2 a b))
+pattern Implies_      s2   a b = Fix (ImpliesPattern (Implies s2 a b))
+pattern In_        s1 s2   a b = Fix (InPattern (In s1 s2 a b))
+pattern Next_         s2   a   = Fix (NextPattern (Next s2 a)) 
+pattern Not_          s2   a   = Fix (NotPattern (Not s2 a))
+pattern Or_           s2   a b = Fix (OrPattern (Or s2 a b))
+pattern Rewrites_     s2   a b = Fix (RewritesPattern (Rewrites s2 a b))
+pattern Top_          s2       = Fix (TopPattern (Top s2))
+pattern Var_             v     = Fix (VariablePattern v)   
```

Comment:
Why didn't you add patterns for strings and chars?

Reply:
Added